symfony/extracting-translations.rst

```diff
@@ -36,3 +36,109 @@ For example, with `FOSUserBundle<https://github.com/FriendsOfSymfony/FOSUserBund
       configs:
         app:
           external_translations_dir: ["%kernel.root_dir%/vendor/friendsofsymfony/user-bundle/Resources/translations"]
+
+
+Create custom extractor
+-----------------------
+
+Example method whose argument we want to translate 
+
+.. code-block:: php
+
+    $this->logger->addMessage("text");
+
+
+Example extractor class that we use to create translations
+
+.. code-block::	php
+
+    <?php
+
+    namespace App\Extractor;
+
+    use PhpParser\Node;
+    use PhpParser\NodeVisitor;
+    use Translation\Extractor\Visitor\Php\BasePHPVisitor;
+    
+    final class MyCustomExtractor extends BasePHPVisitor implements NodeVisitor 
+    {
+           
+        /**
+         * {@inheritdoc}
+        */
+        public function beforeTraverse(array $nodes): ?Node
+        {
+            return null;
+        }
+
+        /**
+        * {@inheritdoc}
+        */
+        public function enterNode(Node $node): ?Node
```

Comment:
Is it OK that this method expects `Node` instance or NULL but we always return only nulls? Don't we want to return `Node` instance at some point?

Reply:
Hi, this code is in large part from Your file
https://github.com/php-translation/extractor/blob/master/src/Visitor/Php/Symfony/FlashMessage.php